### PR TITLE
[VarDumper] upgrade doc to 2.7 wither interface

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.7 |
| Fixed tickets | - |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@nicolas-grekas for this kind of method listings, we no longer use traditional lists in the documentation but "definition lists". The RST syntax in this case would be:

```
:method:`Symfony\\Component\\VarDumper\\Cloner\\Data::withMaxDepth`
    Allows limiting dumps in the depth dimension.
:method:`Symfony\\Component\\VarDumper\\Cloner\\Data::withMaxItemsPerDepth`
    Limits the number of items per depth level.
:method:`Symfony\\Component\\VarDumper\\Cloner\\Data::withRefHandles`
    Allows removing internal objects' handles for sparser output (useful for tests).
    Inlike the previous limits on cloners that remove data on purpose, these can
    be changed back and forth before dumping since they do not affect the
    intermediate representation internally.
```

**Comment by nicolas-grekas** [comment]
updated

File: components/var_dumper/advanced.rst [hunk 1]
```diff
@@ -50,28 +50,33 @@ corresponding Data object could represent only a subset of the cloned variable.
...
+    configures the maximum number of items that will be cloned
+    *past the first nesting level*. Items are counted using a breadth-first
+    algorithm so that lower level items have higher priority than deeply nested
+    items;
+
+:method:`Symfony\\Component\\VarDumper\\Cloner\\VarCloner::setMaxString`
+    configures the maximum number of characters that will be cloned before
+    cutting overlong strings;
+
+In both cases, specifying `-1` removes any limit.
```
Comment (WouterJ) [inline components/var_dumper/advanced.rst]: you have to use a double backtick in reStructured Text to create an inline code block: ``` ``-1`` ```

File: components/var_dumper/advanced.rst [hunk 2]
```diff
@@ -50,28 +50,33 @@ corresponding Data object could represent only a subset of the cloned variable.
...
-method:
-
-* the first ``$maxDepth`` argument allows limiting dumps in the depth dimension,
-* the second ``$maxItemsPerDepth`` limits the number of items per depth level,
-* and the last ``$useRefHandles`` defaults to ``true``, but allows removing
-  internal objects' handles for sparser output,
-* but unlike the previous limits on cloners that remove data on purpose,
-  these can be changed back and forth before dumping since they do not affect
-  the intermediate representation internally.
+:class:`Symfony\\Component\\VarDumper\\Cloner\\Data` object:
```
Comment (WouterJ) [inline components/var_dumper/advanced.rst]: I would prefer "[...] object using the followin methods:"

Comment (nicolas-grekas) [inline components/var_dumper/advanced.rst]: updated

**Comment by weaverryan** [comment]
Thanks Nicolas!

**Pull request merged**